common.py

```diff
@@ -45,74 +46,99 @@ def enum_convert(
             pass
 
         # Finally, try again with all upper case. Also, all "-" are replaced
-        # with "_"
+        # with "_", but only if the result is unique.
         v2 = v.upper().replace("-", "_")
-        for e in enum_type:
-            if e.name.upper() == v2:
-                return e
+        matches = [e for e in enum_type if e.name.upper() == v2]
+        if len(matches) == 1:
+            return matches[0]
 
         raise ValueError(f"Cannot convert {value} to {enum_type}")
 
     raise ValueError(f"Invalid type for conversion to {enum_type}")
 
 
-def enum_convert_list(enum_type: Type[E], input_str: str) -> list[E]:
+def enum_convert_list(enum_type: Type[E], value: Any) -> list[E]:
     output: list[E] = []
 
-    for part in input_str.split(","):
-        part = part.strip()
-        if not part:
-            # Empty words are silently skipped.
-            continue
+    if isinstance(value, str):
+        for part in value.split(","):
+            part = part.strip()
+            if not part:
+                # Empty words are silently skipped.
+                continue
 
-        cases: Optional[list[E]] = None
+            cases: Optional[list[E]] = None
 
-        # Try to parse as a single enum value.
-        try:
-            cases = [enum_convert(enum_type, part)]
-        except Exception:
-            cases = None
+            # Try to parse as a single enum value.
+            try:
+                cases = [enum_convert(enum_type, part)]
+            except Exception:
+                cases = None
 
-        if part == "*":
-            # Shorthand for the entire range (sorted by numeric values)
-            cases = sorted(enum_type, key=lambda e: e.value)
+            if part == "*":
+                # Shorthand for the entire range (sorted by numeric values)
+                cases = sorted(enum_type, key=lambda e: e.value)
 
-        if cases is None:
-            # Could not be parsed as single entry. Try to parse as range.
+            if cases is None:
+                # Could not be parsed as single entry. Try to parse as range.
+
+                def _range_endpoint(s: str) -> int:
+                    try:
+                        return int(s)
+                    except Exception:
+                        pass
+                    return cast(int, enum_convert(enum_type, s).value)
 
-            def _range_endpoint(s: str) -> int:
                 try:
-                    return int(s)
+                    # Try to detect this as range. Both end points may either by
```

Comment:
Please update the docs with the new way of listing test cases.

Reply:
ok. Note that the old way still works, so if you follow the (old) docs, you are still good.

Reply:
README.md adjusted.